x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/planner/PlannerUtils.java

```diff
@@ -294,4 +297,24 @@ public static ElementType toElementType(DataType dataType, MappedFieldType.Field
     public static MappedFieldType.FieldExtractPreference extractPreference(boolean hasPreference) {
         return hasPreference ? DOC_VALUES : NONE;
     }
+
+    /**
+     * Wrap the errors happen during evaluating foldable expressions into {@code VerificationException}.
+     * This is called by {@code EvaluatorMapper.fold()}, {@code ConstantFolding} and {@code PropagateEvalFoldables}.
+     * {@code EvaluatorMapper.fold()} is not aware of the location, null is provided for location.
+     */
+    public static VerificationException convertToVerificationException(Exception e, Location location) {
```

Comment:
Instead of calling this method in `ConstantFolding` and `PropagateEvalFoldables` have you thought about calling it in `Literal.of()` when the `fold()` method is called on the `Literal` constructor?